libraries/Microsoft.Bot.Builder.Runtime/Providers/Adapter/IAdapterProvider.cs

```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Bot.Builder.Runtime.Providers.Adapter
+{
+    public interface IAdapterProvider : IProvider
```

Comment:
comments. 

Is this necessary?  it is an empty interface.  interfaces are hard to extend in a versioning system within c#.  has that been considered here?

Reply:
Documentation added in latest commit. Will evaluate the necessity in another pass tomorrow.

Reply:
On a related note:  Is this Provider/Builder/Extensions framework something we would ever expect devs to extend or modify?  Or is this an implementation detail of Runtime?  I am having a hard time imagining a case where they would need to.  The "public" portion of this is the runtime.json, and all this simply loads that in a platform specific way.

Reply:
The intention is that developers can implement custom Providers and Builders that can be registered as part of our extensibility story and ultimately incorporated via the standard runtime configuration mechanisms (i.e. runtime.json, for now).

The intention behind this is to ensure that the specified provider in the JSON configuration is indeed one intended to register IStorage and related/dependent services with IServiceCollection. I'm not having any immediate thoughts of how else to enforce this constraint outside of an interface without it becoming further convoluted. Open to suggestions, but I think for now I would say this is needed.

Reply:
While allowing the developer to extend this is desirable long term, it's not a requirement in R12/13.  Given the controversial nature of Providers/Builders, it should be marked internal to allow it time to bake and simplify.  Since we are also building other extensibility stories on this (Custom Actions, OnTurnError, etc...), this isn't easily simplified, changed, or eliminated. This will also allow us the space and time to develop additional collateral, like samples and docs, around this story.